source/extensions/filters/http/composite/action.cc

```diff
@@ -9,6 +9,71 @@ void ExecuteFilterAction::createFilters(Http::FilterChainFactoryCallbacks& callb
   cb_(callbacks);
 }
 
+float ExecuteFilterActionFactory::getSampleRatio(
+    const envoy::extensions::filters::http::composite::v3::ExecuteFilterAction& composite_action,
+    Envoy::Runtime::Loader& runtime) {
+  // In default case, if sample_percent is not populated, sample_ratio is 100%, i.e, all sampled.
+  if (!composite_action.has_sample_percent()) {
+    // 1, i.e, 100%,  means all sampled.
+    return 1;
+  }
+  const auto& sample_percent = composite_action.sample_percent().default_value();
+  float denominator = 100;
+  switch (sample_percent.denominator()) {
+  case envoy::type::v3::FractionalPercent::HUNDRED:
+    denominator = 100;
+    break;
+  case envoy::type::v3::FractionalPercent::TEN_THOUSAND:
+    denominator = 10000;
+    break;
+  case envoy::type::v3::FractionalPercent::MILLION:
+    denominator = 1000000;
+    break;
+  default:
+    throw EnvoyException(
+        fmt::format("ExecuteFilterAction sample_percent config denominator setting "
```

Comment:
Exceptions are currently being removed as error reporting mechanism. To make this process easier please use `absl::StatusOr<float>` and only throw at the top level factory method.

Reply:
As this function is called in a couple of places. To avoid throw the exceptions in multiple places, I prefer just throw it here.